src/components/CriteriaProvider/CriteriaProvider.tsx

```diff
@@ -154,6 +167,21 @@ export const CriteriaProvider: FC<CriteriaProviderProps> = memo(({ children }) =
             if (newCriteria) setCriteria(currentCriteria => [...currentCriteria, ...newCriteria]);
           } else if (file.name.endsWith('.cql')) {
             addCqlCriteria(rawContent);
+          } else if (file.name.endsWith('.zip')) {
+            const cqlObj: CqlObject = { main: '', libraries: {} };
+            const zip = await JSZip.loadAsync(file);
+            const zipFiles = Object.values(zip.files);
+            // Check for criteria in each of the cql files, add cql to libraries if no criteria
+            for (let i = 0; i < zipFiles.length; i++) {
+              const fileContents = await zipFiles[i].async('string');
+              const criteria = await cqlToCriteria(fileContents);
```

Comment:
I'm not familiar enough w/ your use case to know why you need to determine the "main" library or what that means, but... I agree that you should send the whole set of files to the translation service.  The service needs access to all referenced libraries in order to translate correctly.

If you need to figure out what the "main" library is, I think your best bet is to analyze the dependencies of each library.  Any library that is _not a dependency of any other libraries_ is a candidate for being the "main" library.  If there is more than one, then I guess you'd need to establish a tie breaker -- maybe of those libraries, the one that itself depends on the most other libraries is the "main" library?  If you can save this analysis for _after_ translation it will be easier to pull out dependencies from the JSON ELM than it would be to pull them from the CQL text.

As for the Authoring Tool having predictably named helper libraries... you can't necessarily count on that.  Authors can upload arbitrary libraries as "External CQL" and those libraries will also be packaged into the zipped download file.

Reply:
Thanks Chris! So after digging into this a little bit, this seems to have started from the pathway execution side of the house, and then that evolved into some incorrect assumptions about how cql-to-elm works. 
In the cql-execution `Library` class, it takes a single ELM object and optionally a Repository containing the dependencies. So there I think we do need to know a "main" library. (unless there's some way of passing them all in together, I admit I haven't looked too closely at that) But so what happened is we built a data model that expected a single main library and a list of dependency libraries.

On the pathway builder side here, we copy & pasted the same cql-to-elm conversion logic & data model because why reinvent the wheel. It may not be exactly the same, but here since we're showing the user individual criteria from the ZIP they uploaded, we still need to identify which library that specific criteria came from, and we can assume the rest are dependencies.

So I do think we can simplify things, though I don't know if it needs to be in this PR specifically:
 - update the cql-to-elm conversion process / data model to take just a list of CQL libraries instead of breaking out a "main" library
 - when we get a list of converted ELM back, loop through each one, and each criteria within it should be associated to the library it came from, with the rest as dependencies

@danlee1025 I'm good with this as-is, but if you think those 2 are easy enough to implement, it might be worth doing that here while we're thinking about it. Then we can get rid of the list of static helper files & related

Reply:
@dehall I think I updated to use this approach, lmk what you think.

Reply:
Yeah looks great. If we really wanted to i think we could inspect the ELM to determine what the actual dependencies are, but that's overkill. 
This works with any zip containing CQL files, not just ones created in the CDS AT, so that's even better than I thought we'd get. 👍